src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -743,55 +743,330 @@ void wm_vuldet_free_report(vu_report *report) {
+            w_strdup(cvss3_vector, report->cvss3->vector_string);
+            report->cvss3->base_score = strtod(cvss3, NULL);
+        }
+        if (!report->cwe) w_strdup(cwe, report->cwe);
+
+        wdb_finalize(stmt);
+
+        // Adding references URLs
+        if (wm_vuldet_prepare(db, vu_queries[VU_REFS_QUERY], -1, &stmt, NULL) != SQLITE_OK) {
+            return SQLITE_ERROR;
+        }
+
+        sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
+        sqlite3_bind_text(stmt, 2, target, -1, NULL);
+
+        ref_count = 0;
+        while (SQLITE_ROW == wm_vuldet_step(stmt)) {
+            reference = (char *)sqlite3_column_text(stmt, 0);
+
+            // Checking if the reference is already in the list
+            int skip = 0;
+            for (int i = 0; report->references && report->references[i]; ++i) {
+                if (!strcmp(reference, report->references[i])) {
+                    skip = 1;
+                    break;
+                }
+            }
+
+            if (skip) continue;
+
+            // The reference is not in the list... adding it.
+            os_realloc(report->references, (ref_count + 2) * sizeof(char *), report->references);
+            os_strdup(reference, report->references[ref_count]);
+            report->references[++ref_count] = NULL;
+        }
+
+        wdb_finalize(stmt);
+
+        // Adding bugzilla references URLs
+        if (wm_vuldet_prepare(db, vu_queries[VU_BUG_REFS_QUERY], -1, &stmt, NULL) != SQLITE_OK) {
+            return SQLITE_ERROR;
+        }
+
+        sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
+        sqlite3_bind_text(stmt, 2, target, -1, NULL);
+
+        while (SQLITE_ROW == wm_vuldet_step(stmt)) {
+            bugzilla_reference = (char *)sqlite3_column_text(stmt, 0);
+            os_realloc(report->bugzilla_references, (bugref_count + 2) * sizeof(char *), report->bugzilla_references);
+            os_strdup(bugzilla_reference, report->bugzilla_references[bugref_count]);
+            report->bugzilla_references[++bugref_count] = NULL;
+        }
+
+        wdb_finalize(stmt);
+
+        // Adding advisories
+        if (wm_vuldet_prepare(db, vu_queries[VU_ADVISORY_QUERY], -1, &stmt, NULL) != SQLITE_OK) {
+            return SQLITE_ERROR;
+        }
+
+        sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
+        sqlite3_bind_text(stmt, 2, target, -1, NULL);
+
+        while (SQLITE_ROW == wm_vuldet_step(stmt)) {
+            advisory = (char *)sqlite3_column_text(stmt, 0);
+            os_realloc(report->advisories, (adv_count + 2) * sizeof(char *), report->advisories);
+            os_strdup(advisory, report->advisories[adv_count]);
+            report->advisories[++adv_count] = NULL;
+        }
+
+        wdb_finalize(stmt);
+    }
+
+    wdb_finalize(stmt);
+
+    return SQLITE_OK;
+}
+
+int wm_vuldet_send_agent_report(sqlite3 *db, OSHash *cve_table, agent_software *agents_it) {
```

Comment:
This is the biggest change where I separate the code of this function to create all the previous functions.